### PR TITLE
Reconciliation loop for TUF resource

### PR DESCRIPTION
**Pull request by osmman**
- configurable service port, external access and secrets
 - autodiscovery of public key and certificate of Rekor and Fulcio
 - reconcile instance when config will change
 - integration test based on Kubebuilder envtest
 
**Run integration tests:**
1. Download the binaries `make envtest`
2. Running tests `make test`

### CODE REVIEW

**Comment by osmman** [comment]
Which version of Go should I target? go.mod file has set 1.21 and CI is using some older version.

**Comment by bouskaJ** [comment]
rebase https://github.com/securesign/secure-sign-operator/blob/main/.github/workflows/main.yml#L10C15-L10C19

**Comment by osmman** [comment]
> rebase https://github.com/securesign/secure-sign-operator/blob/main/.github/workflows/main.yml#L10C15-L10C19

It is not cause of the problem, PR already has this change. It looks like that it comes from https://github.com/securesign/secure-sign-operator/blob/main/Dockerfile#L2

**Comment by osmman** [comment]
Waiting on https://github.com/securesign/secure-sign-operator/pull/42

**Comment by cooktheryan** [comment]
Once @JasonPowr gets the err with ctlog in maybe rebase and lets ship this

**Review by bouskaJ — APPROVED** [review]
I left a few comments but otherwise looks good. Impressive work @osmman :+1:

File: controllers/tuf/tuf_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+		typeNamespaceName := types.NamespacedName{Name: TufName, Namespace: TufNamespace}
+		tuf := &v1alpha1.Tuf{}
+
+		BeforeEach(func() {
+			By("Creating the Namespace to perform the tests")
+			err := k8sClient.Create(ctx, namespace)
+			Expect(err).To(Not(HaveOccurred()))
+
+			By("Setting the Image ENV VAR which stores the Operand image")
+			err = os.Setenv("TUF_IMAGE", "example.com/image:test")
```
[the remark this answers is not in the document]
Comment (osmman) [reply controllers/tuf/tuf_controller_test.go:65]: Just a forgotten residue, I'll remove it

File: config/crd/bases/rhtas.redhat.com_securesigns.yaml [hunk 2]
```diff
@@ -257,6 +262,42 @@ spec:
...
+                          properties:
+                            key:
+                              description: The key of the secret to select from.  Must
+                                be a valid secret key.
+                              type: string
+                            name:
+                              description: 'Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+                                TODO: Add other useful fields. apiVersion, kind, uid?'
+                              type: string
+                            optional:
```
[the remark this answers is not in the document]
Comment (osmman) [reply config/crd/bases/rhtas.redhat.com_securesigns.yaml]: https://github.com/securesign/secure-sign-operator/pull/35/commits/2c8f6f4750fda4b2f8fa191fbde8f4d2a0ac70f5

File: controllers/tuf/finalize.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	// resources that are not owned by this CR, like a PVC.
+
+	// Note: It is not recommended to use finalizers with the purpose of delete resources which are
+	// created and managed in the reconciliation. These ones, such as the Deployment created on this reconcile,
+	// are defined as depended of the custom resource. See that we use the method ctrl.SetControllerReference.
+	// to set the ownerRef which means that the Deployment will be deleted by the Kubernetes API.
+	// More info: https://kubernetes.io/docs/tasks/administer-cluster/use-cascading-deletion/
+
+	// The following implementation will raise an event
+	i.Recorder.Event(instance, "Warning", "Deleting",
```
[the remark this answers is not in the document]
Comment (osmman) [reply controllers/tuf/finalize.go:100]: Yes you are correct, it does not have any meaningful functionality and I unable to image any in the feature for current TUF service. I keep it there as example how to correctly handle finalise into Action design.

[the remark this answers is not in the document]
Comment (osmman) [reply controllers/tuf/finalize.go]: finelizer removed https://github.com/securesign/secure-sign-operator/pull/35/commits/3c6da1c339a650efa76b600bcc147bbd1136dc18

**Pull request merged**